### DISCUSSION TITLE
GraphQL: scheduleOriginId and scheduleSelectorId for stopRunningSchedule mutation

### DISCUSSION THREAD

**Discussion by sburkulemb** [dagster-io/dagster#21025] [Q&A] [answered]
Hey, I need to use the  stopRunningSchedule endpoint and have been getting this. Any help would be appreciated.
![Screenshot 2024-04-04 at 11 08 45](https://github.com/dagster-io/dagster/assets/139243256/6a27cac5-86d4-4272-86bc-464ca3d7f94b)

**Comment by garethbrickman** [comment, marked as answer]
The `scheduleOriginId` and `scheduleSelectorId` need to be IDs for the schedule you want to stop.

 These can be obtained by querying your Dagster instance for the schedules and their states. Here's an example query to fetch schedules:

```graphql
query SchedulesQuery {
  schedulesOrError(repositorySelector: { repositoryName: "your_repository_name", repositoryLocationName: "your_code_location_name" }) {
    ... on Schedules {
      results {
        id
        name
        scheduleState {
          id
          status
        }
      }
    }
    ... on PythonError {
      message
      stack
    }
  }
}
```

This query will return a list of schedules with their `id` (which corresponds to `scheduleSelectorId`) and `scheduleState` (which includes an `id` that corresponds to `scheduleOriginId`). You can then use these IDs in your `stopRunningSchedule` mutation.

Note that:

- repositoryName = name of the Repository (if none, use `"__repository__"`)
- repositoryLocationName = name of the Code Location
- If the Code location name is something like `foo@bar` then the repository name is "foo" and location name is "bar". i.e. `{repository}@{location}`

**Reply by sburkulemb** [reply]
Thanks a lot!
